incubator/drone/templates/deployment-agent.yaml

```diff
@@ -0,0 +1,56 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "fullname" . }}-agent
+  labels:
+    app: {{ template "fullname" . }}-agent
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: {{ template "fullname" . }}-agent
+        chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+        release: "{{ .Release.Name }}"
+        heritage: "{{ .Release.Service }}"
+    spec:
+      containers:
+      - name: {{ template "fullname" . }}-agent
+        image: {{ .Values.image }}
+        imagePullPolicy: {{ default "" .Values.imagePullPolicy | quote }}
+        command:
+          - "/drone"
+          - "agent"
+        env:
+          - name: DRONE_DEBUG
+            valueFrom:
+              configMapKeyRef:
+                name: {{ template "fullname" . }}
+                key: agent.debug.is.enabled
+          {{ if .Values.ssl.enabled }}
+          - name: DRONE_SERVER
+            value: wss://{{ template "fullname" . }}-server/ws/broker
+          {{ else }}
+          - name: DRONE_SERVER
+            value: ws://{{ template "fullname" . }}-server/ws/broker
+          {{ end }}
+          - name: DRONE_SECRET
+            valueFrom:
+              secretKeyRef:
+                name: {{ template "fullname" . }}
+                key: server.secret
+        resources:
+          requests:
+            cpu: 100m
+            memory: 64Mi
+        volumeMounts:
+          # Enables Docker in Docker
+          - mountPath: /var/run/docker.sock
+            name: docker-socket
+      volumes:
+        - name: docker-socket
+          hostPath:
+            path: /var/run/docker.sock
```

Comment:
there's a bug in GKE currently with mounting in the host docker volume with drone: http://discourse.drone.io/t/drone-on-kubernetes-builds-fail-with-docker-errors/162

I would propose that instead we run a privileged container running docker-in-docker as a sidecar container in the pod, which then the agent is configured to point at that engine.